config/kernel-add-disk.m4

```diff
@@ -8,7 +8,7 @@ AC_DEFUN([ZFS_AC_KERNEL_SRC_ADD_DISK], [
 	], [
 		struct gendisk *disk = NULL;
 		int err = add_disk(disk);
-		err = err;
+		return err;
```

Comment:
How about this instead so we're not depending on the return type of the `main()` test function.

```c
int error __attribute__ ((unused)) = add_disk(disk);
```